api_tests/lib_sdk/actors/actor.ts

```diff
@@ -159,28 +160,73 @@ export class Actor {
 
     public async accept() {
         if (!this.swap) {
-            throw new Error("Cannot accept inexistent swap");
+            throw new Error("Cannot accept non existing swap");
         }
 
         await this.swap.accept(Actor.defaultActionConfig);
     }
 
     public async fund() {
         if (!this.swap) {
-            throw new Error("Cannot fund inexistent swap");
+            throw new Error("Cannot fund non existing swap");
         }
 
         this.logger.debug("Funding as part of swap @ %s", this.swap.self);
         await this.swap.fund(Actor.defaultActionConfig);
+
+        const entity = await this.swap.fetchDetails();
+        switch (entity.properties.role) {
+            case "Alice":
+                await this.actors.alice.assertAlphaFunded();
+                await this.actors.bob.assertAlphaFunded();
+                break;
+            case "Bob":
+                await this.actors.alice.assertBetaFunded();
+                await this.actors.bob.assertBetaFunded();
+                break;
+        }
+    }
+
+    public async refund() {
+        if (!this.swap) {
+            throw new Error("Cannot refund non existing swap");
+        }
+
+        this.logger.debug("Refunding as part of swap @ %s", this.swap.self);
+        await this.swap.refund(Actor.defaultActionConfig);
+
+        const entity = await this.swap.fetchDetails();
+        switch (entity.properties.role) {
+            case "Alice":
+                await this.actors.alice.assertAlphaRefunded();
+                await this.actors.bob.assertAlphaRefunded();
```

Comment:
Function called with `await` here but also uses `await` inside the function?

Reply:
That's how it works :D

Reply:
ok, I need a lesson on async/await in TS.  I thought in Singapore you and I used a return statement in order to use await when awaiting on the function.  No response needed, I'll chat with you in person about this some time.